core/kernel/user_ta.c

```diff
@@ -128,6 +128,40 @@ static void dec_recursion(void)
 	tsd->syscall_recursion--;
 }
 
+static bool should_call_utee_ta_entry(enum utee_entry_func func,
+				      uint32_t ta_flags)
+{
+	switch (func) {
+	case UTEE_ENTRY_FUNC_OPEN_SESSION:
+	case UTEE_ENTRY_FUNC_CLOSE_SESSION:
+	case UTEE_ENTRY_FUNC_INVOKE_COMMAND:
+#if defined(CFG_TA_STATS)
+	case UTEE_ENTRY_FUNC_DUMP_MEMSTATS:
+#endif
+		/*
+		 * The function ID can be recognized by TA,
+		 * so it's safe to call __utee_ta_entry().
+		 */
+		return true;
+	default:
+		/*
+		 * Falling to default means the function ID
+		 * can't be recognized by TA.
+		 * if TA header sets the ENTRY_NO_PANIC flag,
+		 * return false to avoid calling __utee_ta_entry()
+		 * where will cause TA panic.
+		 * else, return true to call __utee_ta_entry(),
+		 * get TA panic then.
+		 */
```

Comment:
I think some logic was inverted during reviews.
I TA has NO_PANIC flag set, then it return NOT_SUPPORTED on invalud function ID, right?
So here we should return true when NO_PANIC flag is set.

Another comment `case UTEE_ENTRY_FUNC_DUMP_MEMSTATS` is no different that the other default. The legacy always supported are open/invoke/close. All others require NBO_PANIC flag to be set to be safely called.

I think the fonction could be renamed and should look like:
```c
static bool ta_entry_is_safe(enum utee_entry_func func, uint32_t ta_flags)
{
	switch (func) {
	case UTEE_ENTRY_FUNC_OPEN_SESSION:
	case UTEE_ENTRY_FUNC_CLOSE_SESSION:
	case UTEE_ENTRY_FUNC_INVOKE_COMMAND:
		return true;
	default:
		return ta_flags & TA_FLAG_ENTRY_NO_PANIC;
	}
}